virttest/qemu_devices/qcontainer.py

```diff
@@ -1733,14 +1742,13 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus, iothread,
 
             if sectype == 'password':
                 devices[-1].set_param("file", sec.filename)
-            elif sectype == 'key':
+            elif sectype == 'key' or sectype == 'cookie':
```

Comment:
Looks that when using cookies, the `id` option of the `QObject` should be set with the cookie's `aid`, right?

Reply:
yes, aid is the Cookie object's aid, in _get_access_secrets, both Cookie and StorageAuth objects are collected in a list, which is used here to generate secret QObject objects.